org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/service/ServiceExtension.java

```diff
@@ -62,57 +62,73 @@ public ServiceExtension() {
 	}
 
 	@Override
-	protected boolean supportsType(TargetType targetType, ExtensionContext extensionContext)
+	protected boolean supportsType(TargetType targetType, InjectService injectService,
```

Comment:
This is wrong. Any information about the injection site should be in the TargetType parameter. Passing the annotation is leaky. Improve the information in TargetType if it does not already contain what you need.

Reply:
`ServiceExtension` was the special case here because whether or not it supports a type depends on whether or not the `service` attribute has been set on the annotation.

I managed to work around it by instead moving the validation code from `supportsType()` into `resolveValue()` (which already has access to the annotation instance). This had the bonus of getting rid of the duplicated validation code and not needing the new method which required a version bump.